frontend/src/app/modules/posts/sevices/post.service.ts

```diff
@@ -0,0 +1,34 @@
+import {Injectable, signal} from '@angular/core';
+import {HttpClient, HttpHeaders} from "@angular/common/http";
+import {AddPost} from "../../../shared/types/posts/add-post.model";
+import {finalize, Observable, tap} from "rxjs";
+import {AlertService} from "../../../shared/services/alert.service";
+@Injectable({
+  providedIn: 'root'
+})
+export class PostService {
+
+  constructor(private http: HttpClient, private alertService: AlertService) {
+
+  }
+
+  isLoading = signal(false);
+
+
+  addPost(model: AddPost, userId: string): Observable<void> {
+    this.isLoading.set(true);
+    const headers = new HttpHeaders().set('UserId', userId);
+    new Promise(resolve => setTimeout(resolve, 5000));
```

Comment:
to chyba z testowania ci zostało 🤔 (polecam ustawić w debuggerze chroma throttling na networku generalnie jak testujesz state ładowania)